src/mscorlib/shared/System/Diagnostics/Debug.cs

```diff
@@ -117,6 +135,18 @@ public static void Fail(string message, string detailMessage)
         {
             Assert(false, message, detailMessage);
         }
+        
+        private static string PopStackTrace(string stackTrace)
```

Comment:
Instead of this method, use the StackTrace constructor that lets you specify number of methods to ship. (Take a look at the implementation of EnvironmentAugments.StackTrace.) It will avoid need to reconstruct the stacktrace.